src/radical/pilot/agent/lm/yarn.py

```diff
@@ -160,7 +215,12 @@ def config_yarn_site():
             # Solution to find Java's home folder:
             # http://stackoverflow.com/questions/1117398/java-home-directory
 
-            jpos = subprocess.check_output(['readlink','-f', '/usr/bin/java']).split('bin')
+            java = subprocess.check_output(['which','java'])
+            if java != '/usr/bin/java':
+                jpos=java.split('bin')
+            else:
+                jpos = subprocess.check_output(['readlink','-f', '/usr/bin/java']).split('bin')
```

Comment:
I should motivate the comments: shell callouts can be expensive, and they are not as portable as one would think.